File: SundeeFundee/Models/Enums.swift
import Foundation

enum ExperienceLevel: String, Codable, CaseIterable {
    case beginner
    case intermediate
    case advanced
}

enum PrimaryGoal: String, Codable, CaseIterable {
    case strength
    case hypertrophy
    case explosiveness
}

enum CycleStatus: String, Codable, CaseIterable {
    case active
    case completed
    case paused
}

enum RecordType: String, Codable, CaseIterable {
    case weight
    case volume
}

enum FlowLevel: String, Codable, CaseIterable {
    case light
    case medium
    case heavy
}

enum SymptomCategory: String, Codable, CaseIterable {
    case physical
    case emotional
    case energy
}

enum CyclePhase: String, Codable, CaseIterable {
    case menstrual
    case follicular
    case ovulation
    case luteal
}

enum TimerStatus: String, Codable {
    case idle
    case running
    case paused
    case complete
}

enum NotificationType: String, Codable, CaseIterable {
    case sound
    case vibrate
    case both
    case none
}

File: flutter_app/lib/domain/enums.dart
enum ExperienceLevel { beginner, intermediate, advanced }

enum PrimaryGoal { strength, hypertrophy, explosiveness }

enum CycleStatus { active, completed, paused }

enum RecordType { weight, volume }

enum FlowLevel { light, medium, heavy }

enum SymptomCategory { physical, emotional, energy }

enum CyclePhase { menstrual, follicular, ovulation, luteal }

enum TimerStatus { idle, running, paused, complete }

enum NotificationType { sound, vibrate, both, none }

enum Gender { male, female, preferNotToSay }

extension GenderDisplayName on Gender {
  String get displayName {
    switch (this) {
      case Gender.male:
        return 'Male';
      case Gender.female:
        return 'Female';
      case Gender.preferNotToSay:
        return 'Prefer not to say';
    }
  }
}

enum WeightUnit { lbs, kg }

T enumFromString<T extends Enum>(List<T> values, String? rawValue, T fallback) {
  if (rawValue == null) {
    return fallback;
  }

  for (final value in values) {
    if (value.name == rawValue) {
      return value;
    }
  }

  return fallback;
}
